gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,933 @@
+    "                                  base_name=f\"{k}_norm_sum\",\n",
+    "                                  output_dir=tiff_out_dir,\n",
+    "                                  x_len=324, y_len=324)\n",
+    "        # Print a newline to prevent the saving messages overlapping\n",
+    "        print()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Normalize by open beam\n",
+    "normalized = stitched.copy()\n",
+    "normalized[\"sample\"] = stitched[\"sample\"] / stitched[\"reference\"]\n",
+    "normalized[\"sample_elastic\"] = stitched[\"sample_elastic\"] / stitched[\"reference\"]\n",
+    "normalized[\"sample_plastic\"] = stitched[\"sample_plastic\"] / stitched[\"reference\"]\n",
+    "\n",
+    "# Replace special values nan and inf\n",
+    "replacement=sc.Variable(value=0.0, variance=0.0)\n",
+    "kwargs = {\"nan\" : replacement, \"posinf\" : replacement, \"neginf\" : replacement}\n",
+    "for k in normalized.keys():\n",
+    "    sc.nan_to_num(normalized[k].data, out=normalized[k].data, **kwargs)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "if output_tiff_stack:\n",
+    "    imaging.export_tiff_stack(dataset=normalized, key=\"sample\", x_len=324, y_len=324,\n",
+    "                              base_name=\"initial_tof\", output_dir=os.path.join(output_dir, \"tiffs_tof_initial\"))\n",
+    "    imaging.export_tiff_stack(dataset=normalized, key=\"sample_elastic\", x_len=324, y_len=324,\n",
+    "                              base_name=\"elastic_tof\", output_dir=os.path.join(output_dir, \"tiffs_tof_elastic\"))\n",
+    "    imaging.export_tiff_stack(dataset=normalized, key=\"sample_plastic\", x_len=324, y_len=324,\n",
+    "                          base_name=\"elastic_tof\", output_dir=os.path.join(output_dir, \"tiffs_tof_plastic\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "sc.compat.mantid.load_component_info(normalized, instrument_file)\n",
+    "\n",
+    "normalized.coords[\"detector_mapping\"] = imaging.make_detector_groups(324, 324, nx_target, ny_target)\n",
+    "\n",
+    "wavelength = sc.neutron.convert(normalized, \"tof\", \"wavelength\")\n",
+    "\n",
+    "# Position data ...etc. is dropped on conversion but is later required for plotting\n",
+    "sc.compat.mantid.load_component_info(wavelength, instrument_file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Apply a median filter to the sample after converting to wavelength\n",
```

Comment:
Rewrite this without loops. Similar techniques as I used above for masking may work.